pkg/list/lister.go

```diff
@@ -86,31 +86,44 @@ func (l *lister) GetPackagesWithStatus(
 	index, err := l.fetchRepoAndInstalled(ctx, options, includePackages)
 	result := make([]*PackagesWithStatus, 0, len(index))
 	for _, item := range index {
-		// TODO itemShouldBeIncluded is wrong here – need to check outdated in the inner loop
-		// if itemShouldBeIncluded(&item, options) {
 		ls := make([]*PackageWithStatus, 0, len(item.Packages))
 		for _, pkg := range item.Packages {
-			pkgWithStatus := PackageWithStatus{
-				MetaIndexItem: *item.IndexItem,
-				Package:       pkg,
-				Status:        client.GetStatusOrPending(pkg),
-			}
-			if item.PackageInfo != nil {
-				pkgWithStatus.InstalledManifest = item.PackageInfo.Status.Manifest
+			if packageShouldBeIncluded(&item, pkg, options) {
+				pkgWithStatus := PackageWithStatus{
+					MetaIndexItem: *item.IndexItem,
+					Package:       pkg,
+					Status:        client.GetStatusOrPending(pkg),
+				}
+				if item.PackageInfo != nil {
+					pkgWithStatus.InstalledManifest = item.PackageInfo.Status.Manifest
+				}
+				ls = append(ls, &pkgWithStatus)
 			}
-			ls = append(ls, &pkgWithStatus)
 		}
-		result = append(result, &PackagesWithStatus{
-			MetaIndexItem: *item.IndexItem,
-			Packages:      ls,
-		})
-		// }
+		hasIncludedItems := len(ls) > 0
+		if hasIncludedItems || (!hasIncludedItems && !options.OnlyInstalled && !options.OnlyOutdated) {
```

Comment:
I think this can be simplified
```suggestion
		if hasIncludedItems || (!options.OnlyInstalled && !options.OnlyOutdated) {
```